app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala

```diff
@@ -409,9 +396,28 @@ trait TransactionQueries extends StrictLogging {
     getBalanceQueryDEPRECATED(address).map(sumBalance)
   }
 
-  def getBalanceAction(address: Address): DBActionR[(U256, U256)] = {
-    getBalanceQuery(address).map(sumBalance)
-  }
+  def getBalanceAction(address: Address): DBActionR[(U256, U256)] =
+    getBalanceUntilLockTime(
+      address  = address,
+      lockTime = TimeStamp.now()
+    )
+
+  def getBalanceUntilLockTime(address: Address, lockTime: TimeStamp): DBActionR[(U256, U256)] =
+    sql"""
+      SELECT sum(outputs.amount),
+             sum(CASE
+                     WHEN outputs.lock_time is NULL or outputs.lock_time < ${lockTime.millis} THEN 0
+                     ELSE outputs.amount
+                 END)
```

Comment:
Awesome, I never thought we could do 2 sums and especially using `CASE ....WHEN` like this in it.
Thx SCO (Sql Chief Officer)